src/reducers/secrets.js

```diff
@@ -173,6 +173,15 @@ export function getSecrets(state, namespace) {
   return Object.values(state.byNamespace[namespace] || []);
 }
 
+export function getSecret(state, name, namespace) {
+  const resources = state.byNamespace[namespace] || {};
+  const secretToUse = resources[name];
+  if (secretToUse) {
+    return secretToUse;
+  }
+  return null;
```

Comment:
Can we simplify this?

```suggestion
  return resources[name];
```

if it's not found we'll just return `undefined` which should be fine

Reply:
yep good point, updated now 😸